src/NHibernate/Util/CollectionHelper.cs

```diff
@@ -457,6 +463,11 @@ public object Current
 			#endregion
 		}
 
+		private static class EmptyDictionaryHolder<TKey, TValue>
+		{
+			public static readonly IDictionary<TKey, TValue> Instance = new EmptyMapClass<TKey, TValue>();
```

Comment:
I would move this into `EmptyMapClass` but changed to `internal`.

And then I would introduced an obsolete default public constructor for avoiding future code re-instanciating it directly.
So `EmptyMapClass` would start this way:

```c#
/// <summary>
/// A read-only dictionary that is always empty and permits lookup by <see langword="null" /> key.
/// </summary>
[Serializable]
public class EmptyMapClass<TKey, TValue> : IDictionary<TKey, TValue>
{
#pragma warning disable 618 // Constructor is obsolete, to be switched to non-obsolete but private.
	internal static readonly IDictionary<TKey, TValue> Instance = new EmptyMapClass<TKey, TValue>();
#pragma warning restore 618

	// Since v5.1. To be switched to private.
	[Obsolete("Please use CollectionHelper.EmptyDictionary<TKey, TValue>() instead.")]
	public EmptyMapClass() {}

	private static readonly EmptyEnumerator<TKey, TValue> emptyEnumerator = new EmptyEnumerator<TKey, TValue>();

...
```